### PR TITLE
vk-framework: add geometry map handler

### PR DESCRIPTION
**Pull request by liuyinhangx**
* test-surround-view: fix 'string' missing error in compile stage
 * vk-framework: add geometry map handler
 * vk-copy: support specified copy area
 * test-surround-view: fix double write output in debug mode

### CODE REVIEW

**Comment by liuyinhangx** [comment]
add new patches, and fix issue (https://github.com/intel/libxcam/issues/588), please help to review, thanks.

**Comment by windyuan** [comment]
patch: opencv: fix compile error, please also consider 3.4.3 as maximum version support at this moment.

File: tests/test-surround-view.cpp [hunk 1]
```diff
@@ -180,31 +180,38 @@ add_stream (SVStreams &streams, const char *stream_name, uint32_t width, uint32_
 }
 
 static void
-write_image (const SVStreams &ins, const SVStreams &outs, uint32_t stream_idx)
+write_image (const SVStreams &ins, const SVStreams &outs, uint32_t outs_idx, bool new_frame)
```
Comment (windyuan) [inline tests/test-surround-view.cpp]: a. outs_idx make function complicated. why not use outs[idx] as parameters directly?
b. new_frame doesn't make much sense. suggest separate write_image to write_out_image and write_in_image.

File: tests/test-surround-view.cpp [hunk 2]
```diff
@@ -275,13 +282,13 @@ single_frame (
     while (loop--) {
         CHECK (stitcher->stitch_buffers (in_buffers, outs[0]->get_buf ()), "stitch buffer failed.");
         if (save_output)
-            write_image (ins, outs, 0);
+            write_image (ins, outs, 0, true);
 
         if (save_topview) {
             if (check_stream<SVStreams> (outs, 1)) {
```
Comment (windyuan) [inline tests/test-surround-view.cpp]: also same for check_stream, no need to have idx as "check_stream" parameter

File: tests/test-surround-view.cpp [hunk 3]
```diff
@@ -18,10 +18,11 @@
  * Author: Yinhang Liu <yinhangx.liu@intel.com>
  */
 
+#include <string>
```
Comment (windyuan) [inline tests/test-surround-view.cpp]: move string into test_common.h, we'll widely use it soon

**Comment by windyuan** [comment]
others good. thanks.

**Comment by windyuan** [comment]
merged. thanks.

**Pull request merged**